### PR TITLE
Master fix trendline bugs rar

### PR DESCRIPTION
**Pull request by rrahir**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [4105007](https://www.odoo.com/web#id=4105007&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/helpers/figures/charts/bar_chart.ts [hunk 1]
```diff
@@ -364,6 +364,15 @@ export function createBarChartRuntime(chart: BarChart, getters: Getters): BarCha
      * datasets to ensure the way we distinguish the originals and trendLine datasets after
      */
     trendDatasets.forEach((x) => config.data.datasets!.push(x));
+
+    const originalTooltipTitle = config.options.plugins!.tooltip!.callbacks!.title;
+    config.options.plugins!.tooltip!.callbacks!.title = function (tooltipItem) {
```
Comment (hokolomopo) [inline src/helpers/figures/charts/bar_chart.ts:369]:
```suggestion
    config.options.plugins!.tooltip!.callbacks!.title = function (tooltipItems) {
```

Comment (hokolomopo) [reply src/helpers/figures/charts/bar_chart.ts:369]: same for the other ones

File: src/helpers/figures/charts/chart_common.ts [hunk 2]
```diff
@@ -492,7 +499,14 @@ export function getFullTrendingLineDataSet(
...
   data: number[]
 ) {
-  const backgroundColor = config.color || lightenColor(dataset.backgroundColor as string, 0.5);
+  const defaultBorderColor = colorToRGBA(dataset.backgroundColor as Color);
+  defaultBorderColor.a = 1;
+
+  const borderColor = config.color || lightenColor(rgbaToHex(defaultBorderColor), 0.5);
+  const backgroundRGBA = colorToRGBA(borderColor);
+  // @ts-expect-error
+  if (dataset?.fill) backgroundRGBA.a = LINE_FILL_TRANSPARENCY; // to support area charts
```
Comment (hokolomopo) [inline src/helpers/figures/charts/chart_common.ts:508]: aren't you one of those that tell me that one-liners `if` are ugly and hard to read ? :upside_down_face:

File: tests/figures/chart/chart_plugin.test.ts [hunk 3]
```diff
@@ -2960,4 +2967,56 @@ describe("trending line", () => {
...
+    createChart(
+      model,
+      {
+        type: "line",
+        dataSets: [{ dataRange: "A1", trend: { type: "polynomial", order: 1, display: true } }],
+        fillArea: true,
+      },
+      "chartId"
+    );
+    let runtime = model.getters.getChartRuntime("chartId") as any;
```
Comment (hokolomopo) [inline tests/figures/chart/chart_plugin.test.ts:2983]:
```suggestion
    let runtime = model.getters.getChartRuntime("chartId") as LineChartRuntime;
```

or do we need to do ts-ignore on the rest of the test if we do that  ?

**Review by hokolomopo — COMMENTED** [review]
Runbot is RED :imp: 

Only nitpicks otherwise

File: src/helpers/figures/charts/chart_common.ts (hunk 2 above)
Comment (rrahir) [reply src/helpers/figures/charts/chart_common.ts:508]: <3

File: tests/figures/chart/chart_plugin.test.ts (hunk 3 above)
Comment (rrahir) [reply tests/figures/chart/chart_plugin.test.ts:2983]: We do need the ts ignore because not all runtimes have `chartJsConfig` :/

**Review by LucasLefevre — COMMENTED** [review]
robodoo r+ rebase-ff

**Pull request closed without merging**